spec/models/company_spec.rb

```diff
@@ -9,15 +9,26 @@
     it { is_expected.to validate_presence_of :email }
   end
 
-  describe '.all' do
-    let(:company_a) { double(:company_a, name: 'xyz') }
-    let(:company_b) { double(:company_b, name: 'abc') }
-    let(:company_c) { double(:company_c, name: 'MNO') }
+  describe '#all' do
+    let(:company_a) { double(:company_a, name: 'abc', updated_at: '2016-01-01') }
+    let(:company_b) { double(:company_b, name: 'xyz', updated_at: '2016-01-01') }
+    let(:company_c) { double(:company_c, name: 'MNO', updated_at: '2017-01-01') }
+    let(:company_d) { double(:company_d, name: 'MNO', updated_at: '2017-01-02') }
 
-    it 'sorts by name (ignore case)' do
-      expect(RubysgReboot::COMPANIES).to receive(:map) { [company_a, company_b, company_c] }
+    it 'sorts by updated_at descending, then name ascending (ignore case)' do
```

Comment:
I don't think this test exercises the code it claims, since all the comparisons are stubbed out. 🙂

Reply:
@Drenmi this test describes the expected behavior of the `Company#all` method, which invokes the `<=>` method in the Company model, due to the custom sort order.

Stubbing of the `:<=>` method is needed otherwise this test will fail
```
it 'sorts by updated_at descending, then name ascending (ignore case)' do
  expect(RubysgReboot::COMPANIES).to receive(:map) { [company_a, company_b, company_c, company_d] }
  expect(Company.all).to eq [company_d, company_c, company_a, company_b]
end
```

with the following error:
```
ArgumentError:
    comparison of RSpec::Mocks::Double with RSpec::Mocks::Double failed
```
This is because the test doubles do not respond to the `:<=>` method

Was searching for some examples on how to test sorting with test doubles but did not find anything related, except for [this](https://stackoverflow.com/questions/17220120/rspec-test-comparable). Happy to discuss if there is a more elegant way to test this 😄

Reply:
Right. I didn't see the setup in the diff. 🙂 Instead of doubles, we can just use some `Company` instances:

```ruby
let(:company_a) { Company.new(name: 'abc', updated_on: '2016-01-01') }
```

etc.